### PR TITLE
feat: API to register events and list registered events

### PR DESCRIPTION
**Pull request by alias-rahil**
RESTful API to for events

### CODE REVIEW

File: services/event/db/schema.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+const mongoose = require('mongoose')
+const Joigoose = require('joigoose')(mongoose)
+const Joi = require('joi')
+
+const joiEventSchema = Joi.object({
+  dbType: Joi.string().required(),
```
Comment (thebongy) [inline services/event/db/schema.js]: `dbType` is redundant, can be inferred from `dbURL`. It brings about complications later when a mysql url is provided with a db type of nosql.

File: services/event/db/schema.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  name: Joi.string().required().meta({ unique: true }),
+  dbURL: Joi.string().required(),
+  regCount: Joi.number().required().meta({ default: 0 }),
```
Comment (thebongy) [inline services/event/db/schema.js]: `regCount` can't be a property of an Event, this is unlikely to work structurally in the future

An event can have multiple tables, and the count can be from any one of those tables.

File: services/event/db/schema.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+const mongoose = require('mongoose')
+const Joigoose = require('joigoose')(mongoose)
+const Joi = require('joi')
+
+const joiEventSchema = Joi.object({
```
Comment (thebongy) [inline services/event/db/schema.js]: Rename Event -> Project everywhere

Backends like accounts aren't events

File: services/event/db/schema.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  dbType: Joi.string().required(),
+  name: Joi.string().required().meta({ unique: true }),
+  dbURL: Joi.string().required(),
```
Comment (thebongy) [inline services/event/db/schema.js]: dbURL has to be unique

**Review by thebongy — CHANGES_REQUESTED** [review]

File: services/event/db/schema.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+  regCount: Joi.number().required().meta({ default: 0 }),
+  date: Joi.string().required()
```
Comment (thebongy) [inline services/event/db/schema.js]: Remove this too, it's irrelevant to show the date of the event here, only the placeholder micro service needs to deal with dates

File: services/event/db/schema.js (hunk 2 above)
Comment (alias-rahil) [reply services/event/db/schema.js]: done

File: services/event/db/schema.js (hunk 1 above)
Comment (alias-rahil) [reply services/event/db/schema.js]: done

File: services/event/db/schema.js (hunk 3 above)
Comment (alias-rahil) [reply services/event/db/schema.js]: done

File: services/event/db/schema.js (hunk 4 above)
Comment (alias-rahil) [reply services/event/db/schema.js]: done

File: services/event/db/schema.js (hunk 5 above)
Comment (alias-rahil) [reply services/event/db/schema.js]: done

**Review by thebongy — APPROVED** [review]

**Pull request merged**